src/controllers/profileController.js

```diff
@@ -0,0 +1,127 @@
+ * endpoint entity
+ * @class profileController
+*/
+export default class profileController {
+/**
+   * It receives the user id to search the profiles table for in
+   * the express request onject and returns the user's profile.
+   * @function getProfile
+   * @param {Object} req The express request object
+   * @param {Object} res The express response object
+   * @returns {Object} UserProfile The current user profile
+*/
+  static async getProfile(req, res) {
+    const { params: { userId } } = req;
+    try {
+      const { dataValues: data } = await Profiles.findOne({
+        attributes: { exclude: ['createdat', 'updatedAt'] },
+        where: { userId },
+        include: associationInclude
+      });
+      return successResponse(res, 200, 'Profile fetched successfully', data);
+    } catch (error) {
+      return errorResponse(res, 500, 'Operation not successful. Please try again later.');
+    }
+  }
+  
+  /**
+     * It returns the profiles of all on the platform. It is a function
+     * that is only available to the Super Admin of the platform
+     * @function getAllProfiles
+     * @param {Object} req The express request object
+     * @param {Object} res The express response object
+     * @returns {Array} [UserProfile] An array containing the profiles of all users
+  */
+  static async getAllProfiles(req, res) {
+    try {
+      const { dataValues: data } = await Profiles.findAll({
+        attributes: { exclude: ['createdat', 'updatedAt'] },
+        include: associationInclude
+      });
+      return successResponse(res, 200, 'Profiles fetched successfully', data);
+    } catch (error) {
+      return errorResponse(res, 500, 'Operation not successful. Please try again later.');
+    }
+  }
+  
+  /**
+     * It allows the user to either create or update their profile details
+     * @function upsertProfile
+     * @param {Object} req The express request object
+     * @param {Object} res The express response object
+     * @returns {Object} UserProfile The new profile just created
+  */
+  static async upsertProfile({ body, params, userId }, res) {
+    const {
+      firstName, lastName, department: name, phoneNumber, jobDescription, imageUrl
+    } = body;
+    const id = params.userId || userId;
+    let transaction;
+    try {
+      transaction = await sequelize.transaction();
+      const { dataValues: { id: departmentId } } = await Departments.findOne(
+        { where: { name }, transaction }
+      );
+      const [{ dataValues: userData }, userUpdatedIgnored] = await Users.update(
+        { firstName, lastName }, { where: { id }, transaction }
+      );
+      const [{ dataValues: data }, createdIgnored] = await Profiles.upsert({
+        userId, phoneNumber, departmentId, role: 'Staff', jobDescription, imageUrl
+      }, { returning: true, transaction });
+      data.user = { ...userData };
+      await transaction.commit();
+      return successResponse(res, 200, 'Operation successful', data);
```

Comment:
Hi @darasimiolaifa why do you choose a 200 status code instead of a 201 since you created a resource